internal/builders/container/README.md

```diff
@@ -152,6 +153,115 @@ jobs:
       registry-password: ${{ secrets.GITHUB_TOKEN }}
 ```
 
+#### With GCP Artifact Registry
+
+The following is an example of pushing an image to an [Artifact Registry](https://cloud.google.com/artifact-registry) in GCP and generating the provenance for that image. In order for you to run this example, you will need to have a [Workload Identity Federation](https://cloud.google.com/iam/docs/workload-identity-federation) that enables you to exchange a GitHub token for access within GCP. If you have not yet created one or have not created a provider within your existing federation for GitHub, please review the following resources:
+
+- [Setting up Workload Identity Federation](https://github.com/google-github-actions/auth#setting-up-workload-identity-federation)
+- [Enabling keyless authentication from GitHub Actions](https://cloud.google.com/blog/products/identity-security/enabling-keyless-authentication-from-github-actions)
+
+Once you have a Workload Identity Federation with a GitHub provider, you're ready to begin implementing the GitHub Action below. 
+
+Friendly reminder to set the following environment variables in your GitHub settings:
+```bash
+  # EXAMPLE:
+  # projects/123123412578/locations/global/workloadIdentityPools/my-pool/providers/my-provider
+  WORKLOAD_IDENTITY_PROVIDER=
+  # EXAMPLE:
+  # my-service-account@blank-check-231234.iam.gserviceaccount.com
+  SERVICE_ACCOUNT=
+```
+
+```yaml
+env:
+  IMAGE_NAME: ${{ github.repository }}
+  # FORMAT: 
+  # {region}-docker.pkg.dev/{project-id}/{artifact-registry-name}
+  # EXAMPLE:
+  # northamerica-northeast1-docker.pkg.dev/blank-check-231234/your-repository
+  REPOSITORY_PATH: ${{ vars.REPOSITORY_PATH }}
+
```

Comment:
please set the default pemissions, e.g. 
```yaml
permissions: read-all
```

or 
```yaml
permissions:
   contents: read
```

Reply:
I don't understand the what or why behind this comment. Could you expand please?

Reply:
I think that if this is meant to be a full example it should include a set of default permissions.
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions

The defaults are usually permissive and we'd like to encourage folks to explicitly set them to be less permissive.
https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token